internal/clients/stubs/gcp_stub.go

```diff
@@ -0,0 +1,37 @@
+package stubs
+
+import (
+	"context"
+	"errors"
+
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+)
+
+type gcpCtxKeyType string
+
+var gcpCtxKey gcpCtxKeyType = "gcp-interface"
+var NotImplementedErr = errors.New("stub not yet implemented")
```

Comment:
Nitpick: I would move this into `stubs/errors.go` or something so it's obvious this error is for all stubs.